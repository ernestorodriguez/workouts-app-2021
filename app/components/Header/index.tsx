import React, { ReactElement } from "react";
import { NavLink } from "react-router-dom";

interface HeaderProps {
  internal?: boolean;
}

const Header = ({ internal = false }: HeaderProps): ReactElement => (
  <header className={`main-header ${internal && "main-header--internal"}`}>
    {internal && (
      <NavLink className="back-home" data-js="back-home" exact to="/">
        <img
          className="back-home__icon"
          src="/images/chevron.svg"
          alt="< back"
        />
      </NavLink>
    )}
    <img
      className="main-header__logo"
      data-js="logo"
      src="/images/logo.svg"
      alt="BRAND NAME"
    />
  </header>
);

export default Header;
