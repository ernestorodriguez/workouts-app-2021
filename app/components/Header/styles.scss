.main-header {
  background-color: #ff7f66;
  padding: 20px;
  position: relative;
}

.main-header__logo {
  max-width: 150px;
  margin: 0 auto;
  display: block;
}

.back-home {
  position: absolute;
  left: 0;
  top: 50%;
  margin-top: -25px;
  width: 50px;
  height: 50px;
  padding: 10px;
}

.back-home__icon {
  width: 100%;;
  height: auto;
  fill: white;
}

.main-header--internal {
  .main-header__logo {
    max-width: 100px;
  }
}