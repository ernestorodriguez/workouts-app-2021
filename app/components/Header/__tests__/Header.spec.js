import React from 'react';
import { shallow } from 'enzyme';
import Header from '../index';

describe('Header component', () => {
  test('should match snapshot', () => {
    const component = (
      <Header />
    );
    const wrapper = shallow(component);
    expect(wrapper).toMatchSnapshot();
  });
});
