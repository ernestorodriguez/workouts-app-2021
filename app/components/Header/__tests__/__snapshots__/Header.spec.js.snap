// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component should match snapshot 1`] = `
<header
  className="main-header false"
>
  <img
    alt="BRAND NAME"
    className="main-header__logo"
    data-js="logo"
    src="/images/logo.svg"
  />
</header>
`;
